encoding/basics-and-workflow.md

```diff
@@ -2,36 +2,584 @@
+
+The banding is now less obvious,
+and has been further hidden by adding grain.
+
+Grain is something that some people like,
+and others hate with a passion.
+There are various ways that it can be used,
+ranging from adding a particular mood to a scene
+to preventing other artifacts from showing up.
+It's also common for encoders to
+add their own grain after denoising.
+
+![Fairy Tail — NCOP20 (cleaned)](images/basics_clean.png)
+
+This frame looks fairly clean overall,
+but the encoding may introduce some banding
+in the darker areas of this frame.
+To combat this, we're adding grain
+to just the darker areas.
+
+```Py
+import kagefunc as kgf
+
+grain = kgf.adaptive_grain(deband, strength=0.3, luma_scaling=8)
+```
+![Fairy Tail — NCOP20 (grained)](images/basics_grained.png)
+
+
+
+Another common issue is aliasing.
+Aliasing an artifact that is commonly caused
+by scaling or compression.
+
+![Kimetsu no Yaiba — OP1](images/basics_aliasing.png)
+
+Note the windows to the left.
+You'll want to use anti-aliasing filters
+to deal with this.
+
+```Py
+import vsTAAmbk as taa
+
+aa = taa.TAAmbk(src aatype='Eedi3', opencl=True)
+```
+
+![Kimetsu no Yaiba - OP1 (anti-aliased)](images/basics_anti-aliased.png)
+
+There are a couple of different anti-aliasing filters,
+some faster than others.
+Try out lighter AA if you can first,
+like `Nnedi3`.
+
+
 
 ## Encoding the Result
 
-{% term %}
-$ vspipe.exe script.vpy -y - | x264.exe --demuxer y4m --some example --parameters here --output video.264 -
-{% endterm %}
+Now that the filtering has been applied,
+it's time to encode the result.
+There are two video codecs that are generally accepted nowadays,
+and those are AVC/H264 and HEVC/H265.
+
+For H264 encodes,
+you should be using [x264](x264).
+For x265, [x265](x265).
+Both are also included in ffmpeg,
+but for this guide
+we will be using the standalone programs.
+
+There are various settings you will have to tweak
+to get your desired results.
+
+Here's a couple of common settings:
+* preset
+* keyint
+* crf
+* aq-strength
+* psy-rd
+* output-depth
+* output-csp
+
+The preset sets a lot of settings for you.
+You'll almost always want to run `--preset veryslow` for anime.
+
+The keyint sets how often keyframes will be created.
+For full episodes those are usually
+set to `--min-keyint framerate` and `--keyint framerate*20` or `framerate*30`,
+with "framerate" being the framerate of the video.
+For example, if your video is 23.976fps (most common for anime),
+`min-keyint` would be set to 24 and `keyint` would be set to 480 or 720.
+
+The **C**onstant **R**ate **F**actor sets the constant quality option.
+The default of `--crf 23` is way too low for anime,
+and going that high will result in artifacting popping up
+more often than not.
+For values to use, this varies per encoder.
+The commonly suggested values are as follows (lower is better):
+
+x264:
+* 480p: crf 18 or 19
+* 720p: crf 15 or 16
+* 1080p: crf 16
```

Comment:
That's a first. I've always seen people recommending lower values for lower resolutions.